### ISSUE TITLE
Fails on Ubuntu 17.10

### ISSUE THREAD

**Issue by gylead** [chschnell/build-qt5-rpi#2]
I tried this on my Ubuntu 17.10 installation. Building qtwebengine failed because include file bits/c++config.h is not found (on Ubuntu host include files, not sysroot).

I disabled qtwebengine in configure statement but then the script failed because it tries to apply the patch. Would be nice if the script would check for existence of libQt5WebEngineCore_so_5_9_3.

**Comment by chschnell** [comment]
Thanks for letting me know about this bug, I modified the script to skip the webengine library patch as you suggested, please retry and let me know if it works. In the next couple of days I'll run a test with Ubuntu 17.10 to see about the missing headers. I assume you use 64-Bit Ubuntu?

**Comment by chschnell** [comment]
Ok I've tried it out, here's what I found.

In general, the build host needs its native gcc toolchain (either 32-Bit "i386" or 64-Bit "amd64") and the armhf cross-toolchain in order to build Qt5.

For Qt5Webengine, the build host is additionally required to have an i386 gcc toolchain installed, this is of course always the case if the host architecture is i386.

This means that in order to build Qt5Webengine on a 64-Bit host, you're required to have toolchains for three architectures installed: amd64, i386 and armhf.

The problem is that my instructions for 64-Build hosts no longer work on Ubuntu 17.10, there is a package conflict between armhf and i386. These packages now exclude each other, when you install one you're forced to deinstall the other.

So until I find out how to support the three architectures in Ubuntu 17.10 you're stuck with Debian Stretch if you want to build Qt5Webengine.
